conans/test/utils/tools.py

```diff
@@ -621,7 +621,7 @@ def run(self, command_line, user_io=None, ignore_error=False, should_error=False
             )
             raise Exception(exc_message)
 
-        if should_error and not error:
+        if assert_error and not error:
```

Comment:
This two clauses may be combined into a single one: `if assert_error != error`, and keep the same output for both (changing at least the header `Command failed|not failed`)

Reply:
error might be an integer with different values

Reply:
👍  ok, then `if (assert_error and not error) or (not assert_error and error)` (same as now), but what about merging the two blocks into one?